processor/redactionprocessor/processor.go

```diff
@@ -118,6 +118,23 @@ func (s *redaction) processResourceLog(ctx context.Context, rl plog.ResourceLogs
 		for k := 0; k < ils.LogRecords().Len(); k++ {
 			log := ils.LogRecords().At(k)
 			s.processAttrs(ctx, log.Attributes())
+			s.processLogBody(ctx, log.Body(), log.Attributes())
+		}
+	}
+}
+
+func (s *redaction) processLogBody(ctx context.Context, body pcommon.Value, attributes pcommon.Map) {
+	if body.Type() == pcommon.ValueTypeMap {
+		s.processAttrs(ctx, body.Map())
```

Comment:
I'm not sure the same approach is applicable to the body. 
1. As @gray-oni mentioned, the body map can easily be nested, while it is very rarely the case for regular attributes maps (it goes against otel spec).
2. Adding extra keys like `redaction.masked.keys` in the log body 
  - potentially can invalidate the payload if a particular schema is expected
  - Inconsistent with `redaction.body.masked.count` added to attributes if the body is a string

I would suggest we scan the whole payload, including nested maps, and don't add any extra keys in the body, but use log attributes instead